pkg/api/filter/filter.go

```diff
@@ -0,0 +1,33 @@
+package filter
+
+import "net/http"
+
+// Filter is a wrapper around a http.HandlerFunc.
+type Filter interface {
+	Filter(http.HandlerFunc) http.HandlerFunc
+}
```

Comment:
Can we separate the filter interface and the chain implementation of that interface into separate files? imo it would just help keep things clearer in the mind of the reader.

Reply:
Yeah, that's a good idea. Along with this, I'm going to reintroduce the Chain interface I think just to make it extra clear the differences.